### PR TITLE
Cw/reporter plugin junit

### PR DESCRIPTION
**Pull request by Schwad**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: Gemfile
Comment (james-stocks) [inline Gemfile:28]: We need to run this past @tas50 - I believe we've avoided adding Nokogiri to the ecosystem as it's very difficult to support on all platforms/architectures

Comment (james-stocks) [reply Gemfile:28]: Sorry this is only the `:test` group, whoops!

**Comment by Schwad** [comment]
I've done a rebase here - hopefully good to go after

**Comment by Schwad** [comment]
I think the test failures here are innocuous (but requiring small changes) but will give @clintoncwolfe some time to let us know before this gets tweaked and sent to master

**Comment by Schwad** [comment]
Note: 

this fixes https://github.com/inspec/inspec/issues/5243

**Comment by Schwad** [comment]
It seems that:

locally: 4
CI: 6
Windows CI: 7 (the final failing test)

(I can understand the difference in windows expectation but I'd probably have to dig in for the local v. CI)

```
1) Failure:
--
  | inspec exec with junit formatter::when running the v2 junit reporter::when run on a profile with errors#test_0001_reports failures and errors separately [C:/workdir/test/functional/inspec_exec_junit_test.rb:115]:
  | Expected: "6"
  | Actual: "7"
```

**Comment by mjingle** [comment]
@Schwad Thank you for updating the documentation for this. Once https://github.com/inspec/inspec/pull/5249 is merged, then the documentation file paths will be different and this PR will need to be updated.

**Comment by Schwad** [comment]
Thanks for the headsup @mjingle ! 👍

**Review by Schwad — COMMENTED** [review]
I'd approve this but I'm technically the author (thanks to github) - but this looks ready to rock from me as soon as we readjust those merge conflicts from the aforementioned merge! Thanks all

File: docs/reporters.md
Comment (james-stocks) [inline docs/reporters.md]: Is changing html2 to html a merge problem? I'd think we'd still want to guide people towards html2

File: lib/plugins/inspec-reporter-junit/README.md
Comment (james-stocks) [inline lib/plugins/inspec-reporter-junit/README.md:13]: I feel this documentation could be improved. But since we're addressing a regression with a merge fix we shouldn't hold up on merging - I can raise a follow-up PR with some suggested changes and assign to docs team.

**Comment by james-stocks** [comment]
Looks good to me - currently on master trying to use `junit2` gives `NotImplementedError: 'junit2' is not a valid reporter type.`

On this branch both junit and junit2 are working fine:

```
jstocks@phobos inspec % git checkout cw/reporter-plugin-junit
Branch 'cw/reporter-plugin-junit' set up to track remote branch 'cw/reporter-plugin-junit' from 'origin'.
Switched to a new branch 'cw/reporter-plugin-junit'
jstocks@phobos inspec % bundle exec inspec exec test/fixtures/profiles/basic_profile --reporter=junit 
<?xml version='1.0'?>
<testsuites>
  <testsuite name='basic_profile' tests='1' failed='0' failures='0'>
    <testcase name='a is expected to cmp == &quot;a&quot;' classname='basic_profile.The letter a' target='local://' time='0.000225'/>
  </testsuite>
</testsuites>
jstocks@phobos inspec % bundle exec inspec exec test/fixtures/profiles/basic_profile --reporter=junit2
<?xml version='1.0'?>
<testsuites>
  <testsuite name='basic_profile' tests='1' id='1' errors='0' failures='0' skipped='0' hostname='local://' timestamp='2020-09-22T11:04:10' package='' time='0.000264'>
    <properties/>
    <testcase name='a is expected to cmp == &quot;a&quot;' classname='basic_profile.The letter a' time='0.000264'/>
    <system-out/>
    <system-err/>
  </testsuite>
</testsuites>
jstocks@phobos inspec %
```

**Review by james-stocks — APPROVED** [review]
Branch needs refreshed to solve a merge conflict, but this looks good-to-go to me! Thanks @Schwad

**Pull request merged**
